File: src/main/java/br/com/blz/testjava/dto/request/ProductRequestDTO.java
package br.com.blz.testjava.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
@Builder
public class ProductRequestDTO {

    @NotNull(message = "{validator.product.sku.notnull}")
    @JsonProperty("sku")
    private Integer sku;

    @NotNull(message = "{validator.product.name.notnull}")
    @JsonProperty("name")
    private String name;

    @NotNull(message = "{validator.product.inventory.notnull}")
    @Valid
    @JsonProperty("inventory")
    private InventoryDTO inventory;
}

File: src/main/java/br/com/blz/testjava/product/dto/ProductRequestDTO.kt
package br.com.blz.testjava.product.dto

import br.com.blz.testjava.types.WarehouseType

data class ProductRequestDTO(
  val sku: Long,
  val name: String,
  val inventory: ProductInventoryRequestDTO
)

data class ProductInventoryRequestDTO(
  val warehouses: List<WareHouseRequestDTO>
)

data class WareHouseRequestDTO(
  val locality: String,
  val quantity: Int,
  val type: WarehouseType
)

